ektelo/matrix.py

```diff
@@ -0,0 +1,376 @@
+from ektelo import util
+import numpy as np
+from scipy import sparse
+from scipy.sparse.linalg import LinearOperator, aslinearoperator, lsqr
+from functools import reduce
+
+def _any_sparse(matrices):
+    return any(sparse.issparse(Q.matrix) for Q in matrices)
+
+class EkteloMatrix(LinearOperator):
```

Comment:
I'd like to propose that EkteloMatrix inherit from Marshallable. This would only require adding the line `self.init_params = util.init_params_from_locals(locals())` to the top of the initializer and would enable us to easily serialize instances of EkteloMatrix in the future. There are a couple caveats. 1) We would need to update the methods `prepare_for_json` and `receive_from_json` in util to handle scipy sparse matrices and any other backing matrices passed to the EkteloMatrix initializer. 2) EkteloMatrix needs to be stateless, i.e. none of its members can change after initialization. At first glance this looks to be true already.

Reply:
Yep EkteloMatrix is typically stateless (except in the WorstApprox example we've discussed, at least for now).  

However, some subclasses of EkteloMatrix such as VStack, and Kronecker are defined in terms of other EkteloMatrix objects.  Will Marshallable automatically know how to deal with this recursive dependence?